### PR TITLE
Make `_validate_property` a multilevel method

### PR DESCRIPTION
**Pull request by YuriSizov**
Supersedes a part of https://github.com/godotengine/godot/pull/57654. Makes `_validate_property` a multilevel method, meaning that it will be automatically called on every class in the inheritance chain. Extending classes can override parent behavior.

I didn't touch the hack that https://github.com/godotengine/godot/pull/57654 removed as well, since that was a point of disagreement in the comments. If still desired, this can be fixed with a follow-up PR.

This PR is a bit beefier because I have also renamed every `property` to `p_property` to follow our codestyle.

-----

This PR is a requirement to make `_validate_property` exposed to the API, but it doesn't touch that yet. I'm not sure how to properly pass `PropertyInfo` to the extension API. Probably needs some conversion to a struct and instead of the PropertyInfo being changed in place in the public API the method must return a new struct (or Dictionary in case of GDScript). But I'm not confident in how to make that happen yet.

Still, being multilevel means that it will eventually work with user scripts and extension code.

### CODE REVIEW

**Review by reduz — APPROVED** [review]

**Comment by reduz** [comment]
Looks great, thanks for salvaging this!

**Comment by YuriSizov** [comment]
To avoid conflicts with https://github.com/godotengine/godot/pull/64334, I'll merge this after the other one.

**Comment by YuriSizov** [comment]
Would appreciate a double-check from someone else before this is merged. <3

**Review by akien-mga — APPROVED** [review]
Looks good to me.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Zylann** [comment]
Something this PR revealed is now breaking my builds, in a place where I inherited `GraphNode`.

https://github.com/Zylann/godot_voxel/runs/7964218829?check_suite_focus=true

My guess is that making it multilevel highlighted the fact some classes (like `GraphNode`) implement that method as private. So inheriting `GraphNode` creates that error in the GDCLASS macro somehow, because it tries to access it (the private method), regardless of `_validate_property` actually being used or not in the derived class.

**Comment by YuriSizov** [comment]
Yes, it should be protected. I think I've addressed this with `_property_*_revert`s, but I must've missed it here. Feel free to open a PR.

**Pull request merged**
